app/services/import_cait_indc.rb

```diff
@@ -138,43 +120,55 @@ def import_charts
   end
 
   def import_indicators
-    @indicators.each do |ind|
+    @indicators.
+      reject { |ind| ind[:column_name].match(/_label$/) }.
+      each do |ind|
       CaitIndc::Indicator.create!(indicator_attributes(ind))
     end
   end
 
   def import_labels
-    @legend.
-      map { |l| l.except(:chart_name) }.
-      uniq.
-      each do |ind_v|
-        CaitIndc::Label.create!(label_attributes(ind_v))
+    label_fields = @data.
+      first.
+      keys.
+      grep(/_label$/).
+      map { |l| l.to_s.gsub(/_label$/, '') }
+
+    label_accumulator = []
+    label_fields.each do |lf|
+      @data.each do |d|
+        next if d[:"#{lf}_label"].nil?
+
+        label_accumulator << {
+          indicator_name: lf.to_s,
+          legend_item_name: d[:"#{lf}_label"]
+        }
       end
-  end
+    end
 
-  def import_location_data
-    @data.
-      each do |d|
-        a = location_datum_attributes(d)
-        if a[:location]
-          CaitIndc::LocationDatum.create!(a)
-        else
-          Rails.logger.error "location #{d[:country]} not found. Skipping."
-        end
-      end
+    label_accumulator.uniq.each do |l|
+      CaitIndc::Label.create!(label_attributes(l))
+    end
   end
 
   def import_values
+    ind_keys_no_label = @indicator_keys - @indicator_keys.grep(/_label$/)
     @data.each do |d|
-      location = Location.find_by(wri_standard_name: d[:country])
+      location = Location.find_by(iso_code3: d[:iso])
       unless location
         Rails.logger.error "location #{d[:country]} not found. Skipping."
```

Comment:
Review the way the logger is working so that when running as a rake task to have output showing on the terminal. [consider doing it in a new branch]

Reply:
I'll be leaving this as-is and review it in another branch

Reply:
I'm doing this in a separate branch after all my pending PRs are merged (https://www.pivotaltracker.com/n/projects/2083759)